File: solutions/csharp/squeaky-clean/1/SqueakyClean.cs
public static class Identifier
{
    public static string Clean(string identifier)
    {
        var sb = new System.Text.StringBuilder(identifier.Length);
        bool capitalizeNext = false;
        foreach (char c in identifier)
        {
            if (char.IsControl(c))
            {
                sb.Append("CTRL");
            }
            else if (c == ' ')
            {
                sb.Append('_');
            }
            else if (c == '-')
            {
                capitalizeNext = true;
            }
            else if (!char.IsLetter(c) || c is >= 'α' and <= 'ω') {}
            else
            {
                if (capitalizeNext)
                {
                    sb.Append(char.ToUpperInvariant(c));
                    capitalizeNext = false;
                }
                else
                {
                    sb.Append(c);
                }
            }
        }
        return sb.ToString();
    }
}

File: solutions/java/squeaky-clean/1/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String identifier) {
        boolean capitalize = false;
        StringBuilder result = new StringBuilder();
        for (char c : identifier.toCharArray()) {
            if (c == '-') {
                capitalize = true;
                continue;
            }

            c = switch (c) {
                case ' ' -> '_';
                case '4' -> 'a';
                case '3' -> 'e';
                case '0' -> 'o';
                case '1' -> 'l';
                case '7' -> 't';
                default -> c;
            };

            if (capitalize) {
                c = Character.toUpperCase(c);
                capitalize = false;
            }

            if (c == '_' || Character.isLetter(c)) result.append(c);
        }

        return result.toString();
    }
}
